src/rastervision/tagging/tasks/validation_eval.py

```diff
@@ -18,13 +20,21 @@
 
 class Scores():
     """A set of scores for the performance of a model on a dataset."""
-    def __init__(self, y_true, y_pred, all_tags):
+    def __init__(self, y_true, y_pred, dataset):
         self.f2 = fbeta_score(y_true, y_pred, beta=2, average='samples')
+        self.f2_weighted = fbeta_score(y_true, y_pred, beta=2,
+                                       average='weighted')
+        self.atmos_scores, self.common_scores, self.rare_scores = {},{},{}
         f2_subscores = fbeta_score(y_true, y_pred, beta=2, average=None)
-        self.f2_subscores = dict(zip(all_tags, f2_subscores))
+        for tag in dataset.atmos_tags:
+            self.atmos_scores[tag] = f2_subscores[dataset.all_tags.index(tag)]
+        for tag in dataset.common_tags:
+            self.common_scores[tag] = f2_subscores[dataset.all_tags.index(tag)]
+        for tag in dataset.rare_tags:
+            self.rare_scores[tag] = f2_subscores[dataset.all_tags.index(tag)]
 
     def to_json(self):
-        return json.dumps(self.__dict__, sort_keys=True, indent=4)
+        return json.dumps(self.__dict__, sort_keys=False, indent=4)
```

Comment:
I think you had set this to False back when you were grouping things by the order within the f2_subscores, but we should go back to the old way now that you're grouping them using separate dictionaries.